rtos/TARGET_CORTEX/TARGET_CORTEX_M/mbed_rtx_fault_handler.h

```diff
@@ -53,6 +53,6 @@ typedef struct {
 
 //This is a handler function called from Fault handler to print the error information out.
 //This runs in fault context and uses special functions(defined in mbed_rtx_fault_handler.c) to print the information without using C-lib support.
-__NO_RETURN void mbed_fault_handler (uint32_t fault_type, void *mbed_fault_context_in, void *osRtxInfoIn);
+void mbed_fault_handler (uint32_t fault_type, void *mbed_fault_context_in, void *osRtxInfoIn);
```

Comment:
why are we removing no_return here?

Reply:
@0xc0170 - With the new error handling implementation fault handler will call into error handler which will handle halting the system and thus no longer required here.